common/constants.go

```diff
@@ -176,3 +176,8 @@ const MemoKeyForOperator = "operator"
 
 // ReservedTaskListPrefix is the required naming prefix for any task list partition other than partition 0
 const ReservedTaskListPrefix = "/__cadence_sys/"
+
+type (
+	// ContextKey is the alias type for the context key
+	ContextKey string
```

Comment:
not-sharing a type is the main point of that "don't use a string" lint.  it's better to make all the types unique.

e.g. don't use common.ContextKey elsewhere, make a unique `type contextKey string` like what `scanner.go` does.  then collisions are literally impossible, and you don't have to think about it or look for other uses of the type anywhere in the repo.